File: solutions/java/src/atm/Transaction.java
package atm;

public abstract class Transaction {
    protected final String transactionId;
    protected final Account account;
    protected final double amount;

    public Transaction(String transactionId, Account account, double amount) {
        this.transactionId = transactionId;
        this.account = account;
        this.amount = amount;
    }

    public abstract void execute();
}

File: solutions/csharp/atm/Transaction.cs
namespace ATM
{
    public abstract class Transaction
    {
        protected readonly string transactionId;
        protected readonly Account account;
        protected readonly double amount;

        protected Transaction(string transactionId, Account account, double amount)
        {
            this.transactionId = transactionId;
            this.account = account;
            this.amount = amount;
        }

        public abstract void Execute();
    }
}